components/tools/OmeroPy/test/integration/metadata/test_populate.py

```diff
@@ -180,60 +908,368 @@ def _test_parsing_context(self):
+        fileid = anns[0].file.id.val
         ctx = BulkToMapAnnotationContext(
-            self.client, self.plate, fileid=fileid, cfg=cfg)
+            self.client, target, fileid=fileid, cfg=cfg)
         ctx.parse()
-        assert len(self.get_well_annotations()) == 0
+        assert len(fixture.get_child_annotations()) == 0
 
-        ctx.write_to_omero()
-        was = self.get_well_annotations()
-        assert len(was) == 2
-
-        for ma, wid, wr, wc in was:
-            assert isinstance(ma, MapAnnotationI)
-            assert unwrap(ma.getNs()) == NSBULKANNOTATIONS
-            mv = ma.getMapValueAsMap()
-            assert mv['Well'] == str(wid)
-            assert coord2offset(mv['Well Name']) == (wr, wc)
-            if (wr, wc) == (0, 0):
-                assert mv['Well Type'] == 'Control'
-                assert mv['Concentration'] == '0'
-            else:
-                assert mv['Well Type'] == 'Treatment'
-                assert mv['Concentration'] == '10'
+        if batch_size is None:
+            ctx.write_to_omero()
+        else:
+            ctx.write_to_omero(batch_size=batch_size)
+        oas = fixture.get_child_annotations()
+        assert len(oas) == fixture.annCount
+        fixture.assert_child_annotations(oas)
 
-    def _test_delete_map_annotation_context(self):
+    def _test_delete_map_annotation_context(self, fixture, batch_size):
         # self._test_bulk_to_map_annotation_context()
-        assert len(self.get_well_annotations()) == 2
+        assert len(fixture.get_child_annotations()) == fixture.annCount
+
+        cfg = fixture.get_cfg()
 
-        ctx = DeleteMapAnnotationContext(self.client, self.plate)
+        target = fixture.get_target()
+        ctx = DeleteMapAnnotationContext(self.client, target, cfg=cfg)
         ctx.parse()
-        assert len(self.get_well_annotations()) == 2
+        assert len(fixture.get_child_annotations()) == fixture.annCount
+
+        if batch_size is None:
+            ctx.write_to_omero()
+        else:
+            ctx.write_to_omero(batch_size=batch_size)
+        assert len(fixture.get_child_annotations()) == 0
+        assert len(fixture.get_all_map_annotations()) == 0
+
+
+@mark.skipif(sys.version_info < (2, 7),
+             reason="requires python2.7")
+class TestPopulateMetadataHelperPerMethod(TestPopulateMetadataHelper):
+
+    # Some tests in this file check the counts of annotations in a fixed
+    # namespace, and therefore require a new client for each test method
+
+    def setup_class(cls):
+        pass
+
+    def teardown_class(cls):
+        pass
+
+    def setup_method(self, method):
+        super(TestPopulateMetadataHelperPerMethod, self).setup_class()
+
+    def teardown_method(self, method):
+        super(TestPopulateMetadataHelperPerMethod, self).teardown_class()
+
+
+@mark.skipif(sys.version_info < (2, 7),
+             reason="requires python2.7")
+class TestPopulateMetadata(TestPopulateMetadataHelper):
```

Comment:
My reading of https://docs.pytest.org/en/latest/skipping.html#id1 is that the function (or class in that case) is initialized and then the abortion is testing when setting up the tests. Here, because there is some Python 2.7 specific bit in the static class variables, the class initialization fails before it has a chance to execute the skipping logic.